diff --git a/node_modules/mineflayer/lib/plugins/entities.js b/node_modules/mineflayer/lib/plugins/entities.js
index 666b0b3..8b3bba1 100644
--- a/node_modules/mineflayer/lib/plugins/entities.js
+++ b/node_modules/mineflayer/lib/plugins/entities.js
@@ -271,7 +271,11 @@ function inject (bot) {
     entity.pitch = conv.fromNotchianPitchByte(packet.pitch)
     entity.headPitch = conv.fromNotchianPitchByte(packet.headPitch)
 
-    const notchVel = new Vec3(packet.velocityX, packet.velocityY, packet.velocityZ)
+    const notchVel = new Vec3(
+      packet.velocityX ?? packet.velocity?.x,
+      packet.velocityY ?? packet.velocity?.y,
+      packet.velocityZ ?? packet.velocity?.z
+    )
     entity.velocity.update(conv.fromNotchVelocity(notchVel))
     entity.metadata = parseMetadata(packet.metadata, entity.metadata)
 
@@ -281,7 +285,11 @@ function inject (bot) {
   bot._client.on('entity_velocity', (packet) => {
     // entity velocity
     const entity = fetchEntity(packet.entityId)
-    const notchVel = new Vec3(packet.velocityX, packet.velocityY, packet.velocityZ)
+    const notchVel = new Vec3(
+      packet.velocityX ?? packet.velocity?.x,
+      packet.velocityY ?? packet.velocity?.y,
+      packet.velocityZ ?? packet.velocity?.z
+    )
     entity.velocity.update(conv.fromNotchVelocity(notchVel))
   })
 
